packages/app/src/cli/models/extensions/extension-instance.ts

```diff
@@ -115,8 +119,8 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
     this.directory = options.directory
     this.specification = options.specification
     this.devUUID = `dev-${randomUUID()}`
-    this.localIdentifier = basename(options.directory)
-    this.idEnvironmentVariableName = `SHOPIFY_${constantize(basename(this.directory))}_ID`
+    this.localIdentifier = this.configuration.handle ? this.configuration.handle : basename(options.directory)
```

Comment:
We will need a different fallback for a localIdentifier when a handle isn't present, otherwise both extensions will share the same one.